### ISSUE TITLE
--deny-namespaces skips *v1.Status events

### ISSUE THREAD

**Issue by s-urbaniak** [prometheus-operator/prometheus-operator#3410]
**What happened?**

During an unrelated experiment https://github.com/prometheus-operator/prometheus-operator/pull/3409 it occurred to me that the existing deny listing watch logic which filters namespaced objects only can miss Watch events, i.e. `*v1.Status` event object types.

https://github.com/prometheus-operator/prometheus-operator/blob/5c3520960e179458c0c98dae94d4cae65f71aa7f/pkg/listwatch/namespace_denylist.go#L148-L154

I am not fully aware of the functionality ramifications, but currently, if the user specifies a denylist of namespaces, potentially status Watch events will never be dispatched.

**Did you expect to see something different?**

I expect all events related to denylisted namespaces to be dispatched.

**How to reproduce it (as minimally and precisely as possible)**:

Set up prometheus-operator with `--deny-namespaces` and watch for log entries like:

```
msg="unexpected object type in event (*v1.Status): &Status{ListMeta:ListMeta{SelfLink:,ResourceVersion:,Continue:,RemainingItemCount:nil,},Status:Failure,Message:too old resource version: 30297 (42339),Reason:Expired,Details:nil,Code:410,}"
```

**Environment**

* Prometheus Operator version: 0.40.0

* Kubernetes version information: 1.19 (but proably earlier ones too)

* Kubernetes cluster kind: N/A

* Manifests: N/A

* Prometheus Operator Logs: See above

**Anything else we need to know?**:

**Comment by s-urbaniak** [comment]
suggestion: let's refactor the denylist logic as a follow-up to the planned refactoring of multilistwatcher.

We learned that providing a custom `ListWatcher` interface always causes unexpected side-effects with the underlying client-go machinery as experienced in [1] and [2].

[1] https://github.com/prometheus-operator/prometheus-operator/issues/3218
[2] https://github.com/prometheus-operator/prometheus-operator/issues/3218

**Comment by s-urbaniak** [comment]
Continued investigation showed that we can leverage `FieldSelectors` (see https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/) in the style of `namespace!=foo,namespace!=bar,...` to implement the same functionality as the existing denylisting filter.

It also benefits from increased performance as the resultset is limited in the first place, without the need for in-place filtering.

**Comment by s-urbaniak** [comment]
This is fixed in #3440, closing.
